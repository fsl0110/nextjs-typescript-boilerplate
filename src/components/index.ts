// this kind of reexporting may cause problems.
// see: https://nextjs.org/docs/messages/export-all-in-page

export * from './Button/Button';
export * from './Header/Header';
export * from './FormErrors/FormErrors';
