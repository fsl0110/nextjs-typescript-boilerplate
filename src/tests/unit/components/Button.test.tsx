import { render } from 'tests';
import { Button } from 'components';

describe('Button', () => {
  it('renders Button component', () => {
    render(<Button label="My Button" />);
    /*     screen.debug(); */
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Button label="My Button" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
