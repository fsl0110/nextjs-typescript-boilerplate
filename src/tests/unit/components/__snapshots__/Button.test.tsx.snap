// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    My Button
  </button>
</DocumentFragment>
`;
